File: usecase-ensured/src/main/java/com/github/usecase_ensured/data/Request.java
package com.github.usecase_ensured.data;

import io.restassured.http.Header;

import java.util.List;

public record Request(
        Method method,
        List<Header> headers,
        String url,
        String body
) {

    public enum Method {
        GET,
        POST,
        PUT,
        DELETE
    }
}

File: lib/src/main/kotlin/com/github/usecase_assured/data/Request.kt
package com.github.usecase_assured.data

import io.restassured.http.Header

internal data class Request(
    var method: Method,
    var headers: List<Header>,
    var url: String,
    var body: String?
) {

    internal enum class Method {
        GET,
        POST,
        PUT,
        DELETE
    }
}
